aten/src/ATen/native/AffineGridGenerator.cpp

```diff
@@ -0,0 +1,84 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+namespace at { namespace native {
+
+at::Tensor linspace_from_neg_one(const Tensor& grid, int64_t num_steps) {
+  if (num_steps > 1) {
+    return at::linspace(-1, 1, num_steps, grid.options());
+  } else {
+    return at::full(-1, {}, grid.options());
+  }
+}
+
+Tensor make_base_grid_4D(const Tensor& theta, int64_t N, int64_t C, int64_t H, int64_t W) {
+  auto base_grid = at::empty({N, H, W, 3}, theta.options());
+
+  base_grid.select(-1, 0) = linspace_from_neg_one(theta, W);
```

Comment:
Does this actually work? This creates a `Tensor` and then assigns it `linspace_from_neg_one(theta, W)`. What you really want is `base_grid.select(-1, 0).copy_(linspace_from_neg_one(theta, W)`

Reply:
I think it does actually work, the test failures were caused by changing:
`    return at::empty({}, grid.options()).fill_(-1);` 
to
`    return at::full(-1, {}, grid.options());     `

Reply:
Ed says (I am paraphrasing) "this does work because `base_grid.select(-1, 0)` gives an rvalue reference on a tensor and we have an overload on the assignment operator to do a `copy_`"

@eellison could you make it more explicit by writing `base_grid.select(-1, 0).copy_(linspace_from_neg_one(theta, W));`? This is equivalent to what is in the code right now but it is easier to reason about.